.github/workflows/codeql-analysis.yml

```diff
@@ -38,11 +38,11 @@ jobs:
 
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v3
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # tag=v3.0.2
 
       # Initializes the CodeQL tools for scanning.
       - name: Initialize CodeQL
-        uses: github/codeql-action/init@v2
+        uses: github/codeql-action/init@3f62b754e23e0dd60f91b744033e1dc1654c0ec6 # tag=v2
```

Comment:
I agree we should probably be consistent with how we use the full semver tag elsewhere. But now that you mention it and since it's renovate-bot uses these tags I'm now wondering what kind of difference it makes for the update PRs it sends. If for example it pins to a specific version and doesn't send upgrade PRs that may not actually be what we want. Need to look more at the docs...

Reply:
I think there are two goals we have for these comments:

1. Provide an easy to read comment that says what exact released version the sha is taken from.
2. Provide a hint to renovate-bot to tell it to stick to tagged releases and not send us PR to update to sha's at HEAD
 
[Renovate's docs](https://docs.renovatebot.com/modules/manager/github-actions/#additional-information) say they "follow" the provided tag. For us, I think the point of using it was to have it send PRs to update sha's automatically but if we pin to specific semver versions, it may not ever do that? And even if it did, we would never want to accept a PR like that.

And even if we stick to a `v2` tag in order to be able to update to later tags, we will be dependent on those repos maintaining a `v2` tag or similar, and still won't know exactly what version that our sha is referencing.

Maybe @rarkins knows more and has more suggestions.

Reply:
If you follow a major stream like that then you'll still get PRs updating the digest but you won't know exactly which version it is (eg 2.1.2) without manually looking it up. You won't save any PRs either, so in general you're better to use the full version and get better visibility